pragma-node/src/handlers/entries/mod.rs

```diff
@@ -77,22 +78,59 @@ pub struct GetVolatilityResponse {
 pub struct GetOnchainParams {
     pub network: Network,
     pub aggregation: Option<AggregationMode>,
-    pub timestamp: Option<u64>,
     pub routing: Option<bool>,
+    #[serde(default, with = "timestamp_serde")]
+    pub timestamp: Option<Timestamp>,
+}
+
+#[derive(Debug, Clone)]
+pub enum Timestamp {
+    Single(u64),
+    Range(RangeInclusive<u64>),
+}
+
+impl From<u64> for Timestamp {
+    fn from(ts: u64) -> Self {
+        Timestamp::Single(ts)
+    }
+}
+
+mod timestamp_serde {
```

Comment:
why `mod timestamp_serde` instead of `impl<'de> Deserialize<'de> for Timestamp {` ? I'm fine with it but just don't understand why only for this type we would separate this logic in `mod`

Reply:
less code and easier to implement/read but i can change

Reply:
yeah let's use the impl only please

Reply:
done